src/quacc/atoms/skzcam.py

```diff
@@ -1,35 +1,180 @@
 from __future__ import annotations
 
+from importlib.util import find_spec
 from pathlib import Path
 from typing import TYPE_CHECKING
 
 import numpy as np
 from ase.atoms import Atoms
 from ase.data import atomic_numbers
-from ase.io import write
+from ase.io import read, write
 from ase.units import Bohr
+from monty.dev import requires
 from monty.io import zopen
 from monty.os.path import zpath
 
 if TYPE_CHECKING:
     from numpy.typing import NDArray
 
+has_chemshell = find_spec("chemsh") is not None
+
+
+def get_cluster_info_from_slab(
+    adsorbate_slab_file: Path, slab_center_idx: list[int], adsorbate_idx: list[int]
+) -> tuple[Atoms, Atoms, int, NDArray, NDArray]:
+    """
+    Read the file containing the periodic slab and adsorbate (geometry optimized) and return the key information needed to create an embedded cluster in ChemShell.
+
+    Parameters
+    ----------
+    adsorbate_slab_file
+        The path to the file containing the adsorbate molecule on the surface slab. It can be in any format that ASE can read.
+    adsorbate_idx
+        The indices of the atoms that make up the adsorbate molecule.
+    slab_center_idx
+        The the adsorbate_slab_file, this gives the indices of the atoms that are at the 'center' of the slab right beneath the adsorbate.
+
+    Returns
+    -------
+    Atoms
+        The Atoms object of the adsorbate molecule.
+    Atoms
+        The Atoms object of the surface slab.
+    int
+        The index of the first atom of the slab as listed in slab_center_idx.
+    NDArray
+        The position of the center of the cluster.
+    NDArray
+        The vector from the center of the slab to the center of mass of the adsorbate.
+    """
+
+    # Get the necessary information for the cluster from a provided slab file (in any format that ASE can read)
+    adsorbate_slab = read(adsorbate_slab_file)
+
+    # Find indices (within adsorbate_slab) of the slab
+    slab_idx = [i for i, _ in enumerate(adsorbate_slab) if i not in adsorbate_idx]
+
+    # Create slab from adsorbate_slab
+    slab = adsorbate_slab[slab_idx]
+
+    # Find index of the first center atom of the slab as listed in slab_center_idx
+    slab_first_atom_idx = next(
+        index for index, x in enumerate(slab_idx) if x == slab_center_idx[0]
+    )
+
+    # Get the centre of the cluster from the atom indices
+    slab_center_position = np.zeros(3)
+    for atom_idx in slab_center_idx:
+        slab_center_position += adsorbate_slab.get_positions()[atom_idx]
```

Comment:
You generally never need to pre-initialize an array or list in Python.

Here, you can probably do something like

```python
adsorbate_slab[slab_center_idx].get_positions()
```

and use that directly.